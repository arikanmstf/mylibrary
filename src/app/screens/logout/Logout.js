/**
 * Screen Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import * as React from 'react';
import logger from 'helpers/logger';
import type { LogoutProps } from './LogoutTypes';

class Logout extends React.Component<LogoutProps> {
  componentDidMount() {
    const { makeLogoutRequest } = this.props;
    makeLogoutRequest();
  }

  render() {
    logger.log('render: Logout');
    return null;
  }
}

export default Logout;
