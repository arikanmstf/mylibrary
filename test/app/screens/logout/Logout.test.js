/**
 * Screen Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import Logout from 'screens/logout/Logout';

describe('test/app/screens/logout/Logout.test.js', () => {
  const props = {
    makeLogoutRequest: jest.fn(),
  };

  it('Render', () => {
    const wrapper = shallow(<Logout {...props} />);
    const { makeLogoutRequest } = props;

    expect(wrapper).toMatchSnapshot();
    expect(makeLogoutRequest).toBeCalled();
  });
});
