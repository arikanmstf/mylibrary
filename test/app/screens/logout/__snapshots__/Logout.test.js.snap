// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/screens/logout/Logout.test.js Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logout
    makeLogoutRequest={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
